### PR TITLE
TST: #31922 assert no segmentation fault with numpy.array.__contains__

### LINKED ISSUE #31922: BUG: segmentation fault with pd.NA and np.ndarray.__contains__
Testing for membership in an ndarray segfaults:

```
In [1]: pd.NA in np.array(["a"], dtype=object) 
Segmentation fault (core dumped)
```

Now, this might be something that needs to be fixed in numpy, as for plain lists it "works":

```
In [3]: pd.NA in ['a']  
...
TypeError: boolean value of NA is ambiguous
```

but numpy might not expect an error instead of True/False.

### PR DESCRIPTION
**Pull request by ylin00**
- [ ] closes #31922
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] new test cases

### CODE REVIEW

File: pandas/tests/scalar/test_na_scalar.py [hunk 1]
```diff
@@ -305,3 +305,17 @@ def test_pickle_roundtrip_containers(as_frame, values, dtype):
...
+    msg = "boolean value of NA is ambiguous"
+    with pytest.raises(TypeError, match=msg):
+        NA in np.array(["a"], dtype=object)
+
+
+def test_list_contains_na():
+    # GH 31922
+    msg = "boolean value of NA is ambiguous"
+    with pytest.raises(TypeError, match=msg):
+        NA in ["a"]
```
Comment (mroeschke) [inline pandas/tests/scalar/test_na_scalar.py:321]: Could you just make 1 test and parameterize over `np.array(["a"], dtype=object)` and `["a"]`?

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Could you merge master again? Should fix the CI failures

**Comment by jreback** [comment]
thanks @ylin00

**Comment by jreback** [comment]
see comments on the issues. this appears to be failing on older numpies; its ok to skip those in the CI.

**Pull request merged**